restaking/src/solidity_interfaces/IBaseMiddlewareReader.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+interface IBaseMiddlewareReader {
+    function getCaptureTimestamp() external view returns (uint48 timestamp);
+
+    function stakeToPower(address vault, uint256 stake) external view returns (uint256 power);
+
+    function keyWasActiveAt(uint48 timestamp, bytes memory key) external view returns (bool);
+
+    function operatorKey(
+        address operator
+    ) external view returns (bytes memory);
+
+    function operatorByKey(
+        bytes memory key
+    ) external view returns (address);
+
+    function NETWORK() external view returns (address);
+
+    function SLASHING_WINDOW() external view returns (uint48);
+
+    function VAULT_REGISTRY() external view returns (address);
+
+    function OPERATOR_REGISTRY() external view returns (address);
+
+    function OPERATOR_NET_OPTIN() external view returns (address);
+
+    function operatorsLength() external view returns (uint256);
+
+    function operatorWithTimesAt(
+        uint256 pos
+    ) external view returns (address, uint48, uint48);
+
+    function activeOperators() external view returns (address[] memory);
+
+    function activeOperatorsAt(
+        uint48 timestamp
+    ) external view returns (address[] memory);
```

Comment:
to check symbiotic active operators, you use `timestamp` where eigen uses `blockNumber`.

Reply:
To get operators weight, it would require to either do a request for each Operator or deploy a smart contract with a specific function that does everything in one call (Just like Eigen OperatorStateRetriever is doing).
@patrick-ogrady what do you think about everything? I think at the moment the `constructor` and even the semantic between `blockNumber` and `timestamp` are two much distant to create a common trait.
Do we want to clean up and merge stuff in a commonware-restaking lib? Or we need to answer more questions we have?

Reply:
Did some updates trying to cleanup the build. I have to generate files in the `src` instead of `$OUT_DIR` because I cannot pass a concatenated path (`concat!(env!("OUT_DIR"), "......"`) using a non literal string to the `sol!` macro.

Reply:
> I think at the moment the constructor and even the semantic between blockNumber and timestamp are two much distant to create a common trait.

I think the worst thing is to be "too clever" and create a standard interface with a ton of subtle assumptions.

I'd be willing to just do `restaking::{symbiotic,eigenlayer}` for now with no shared interface.

Reply:
> Do we want to clean up and merge stuff in a commonware-restaking lib? Or we need to answer more questions we have?

I need to take a deeper pass on both. Apologize for the delay. Didn't think you'd get everything together this fast lol 😅 .

Reply:
@patrick-ogrady don't worry ! I know you have lot of things to do 🙂 
I think the cleanest PR right now is this one, it reflects what I currently have in mind for the Eigenlayer PR (file organization, build, ...).

Reply:
Thanks for the context!